pyspeckit/cubes/tests/test_cubetools.py

```diff
@@ -6,32 +6,94 @@
 
 import os
 
-def test_subimage_integ_header():
+def make_test_cube(shape=(30,9,9), outfile='test.fits', sigma=None, seed=0):
+    """
+    Generates a simple gaussian cube with noise of
+    given shape and writes it as a fits file.
+    """
+    from astropy.convolution import Gaussian1DKernel, Gaussian2DKernel
+    if sigma is None:
+        sigma1d, sigma2d = shape[0]/10., np.mean(shape[1:])/5.
+    else:
+        sigma1d, sigma2d = sigma
+
+    gauss1d = Gaussian1DKernel(stddev=sigma1d, x_size=shape[0])
+    gauss2d = Gaussian2DKernel(stddev=sigma2d, x_size=shape[1], 
+                                               y_size=shape[2])
+    test_cube = gauss1d.array[:,None,None] * gauss2d.array
+    test_cube=test_cube/test_cube.max()
+    # adding noise:
+    np.random.seed(seed)
+    noise_cube = (np.random.random(test_cube.shape)-.5)* \
+                        np.median(test_cube.std(axis=0))
+    test_cube += noise_cube
+    true_rms = noise_cube.std()
+    
+    # making a simple header for the test cube:
+    test_hdu = fits.PrimaryHDU(test_cube)
+    # the strange cdelt values are a workaround 
+    # for what seems to be a bug in wcslib:
+    # https://github.com/astropy/astropy/issues/4555
+    cdelt1, cdelt2, cdelt3 = -(4e-3+1e-8), 4e-3+1e-8, -0.1
+    keylist = {'CTYPE1': 'RA---GLS', 'CTYPE2': 'DEC--GLS', 'CTYPE3': 'VRAD',
+               'CDELT1': cdelt1, 'CDELT2': cdelt2, 'CDELT3': cdelt3,
+               'CRVAL1': 0, 'CRVAL2': 0, 'CRVAL3': 5, 
+               'CRPIX1': 9, 'CRPIX2': 0, 'CRPIX3': 5,
+               'CUNIT1': 'deg', 'CUNIT2': 'deg', 'CUNIT3': 'km s-1',
+               'BUNIT' : 'K', 'EQUINOX': 2000.0}
+    # write out some values used to generate the cube:
+    keylist['SIGMA' ] = abs(sigma1d*cdelt3), 'in units of CUNIT3'
+    keylist['RMSLVL'] = true_rms
+    keylist['SEED'  ] = seed
+
+    try:
+        test_header = fits.Header(keylist)
+        assert test_header[list(keylist)[0]] == keylist[list(keylist)[0]]
+    except AssertionError:
+        # generates an empty header on some systems (TODO: why?)
```

Comment:
Okay, I can't get the fits.Header(dict) to work correctly anywhere now. Tested it for astropy versions 1.1.dev14278 and 1.1.1. I was sure it worked before, but maybe I just ran some tests on a fits file with an empty header. Looking at the [source file](https://github.com/astropy/astropy/blob/master/astropy/io/fits/header.py) for Header class I don't think the dictionary values are ever being stored in the init function.